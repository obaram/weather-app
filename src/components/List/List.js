import React from 'react';
import {Listitem} from './ListItem/Listitem'
import { Table, Alert} from 'react-bootstrap';
import styles from './List.module.scss'

const List = ({items,handleDelete,unit,error, success}) => (

    <>
    {error.show && <Alert  className={styles.alert} key="alert" variant="warning">{error.msg}</Alert>  }
    {success.show && <Alert  className={styles.alert} key="success" variant="success">{success.msg}</Alert>  } 
   
   <Table striped bordered hover className={styles.wrapper}>
    <thead>
        <tr>
        <th>#</th>
        <th>Miasto</th>
        <th>Åšrednia prognozowana tempretura</th>
        <th></th>
        </tr>
    </thead>
    <tbody>
        
        {
            items.map((item,index) => <Listitem item={item} index={index} handleDelete={handleDelete} key={index} unit={unit}/>)
        }
        
    </tbody>
    </Table>
    </>
)

export default List;
