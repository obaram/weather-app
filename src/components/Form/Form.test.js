import React from 'react';
import ReactDOM from 'react-dom';
import List from '../List/List';
import { shallow } from 'enzyme';
import renderer from 'react-test-renderer'; 
import {BrowserRouter,Route, Switch} from 'react-router-dom'
import "../../../setupTest"
import AddForm from "./Form";

 

  it('renders correctly', () => {
    const tree = renderer.create(<AddForm />).toJSON();
    expect(tree).toMatchSnapshot();
  });


 describe('ListItem', () => {
     it('should render list item',()=>{
     
     const wrapper = shallow(<AddForm/>);
     expect(wrapper).toMatchSnapshot();

     })

})
