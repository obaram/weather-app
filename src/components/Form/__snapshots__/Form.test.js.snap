// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Form.js should render list item 1`] = `
<ContextConsumer>
  <Component />
</ContextConsumer>
`;

exports[`ListItem should render list item 1`] = `
<ContextConsumer>
  <Component />
</ContextConsumer>
`;

exports[`renders correctly 1`] = `
<form
  className="wrapper"
  onSubmit={[Function]}
>
  <div
    className="items form-group"
  >
    <input
      autoComplete="off"
      className="text form-control"
      id="addCityForm"
      onChange={[Function]}
      placeholder="Podaj miasto"
      type="text"
    />
    <button
      className="btn btn btn-primary"
      disabled={false}
      type="submit"
    >
      Dodaj
    </button>
  </div>
</form>
`;
